src/platform/sdl2/sdl2mouse.cpp

```diff
@@ -162,34 +159,32 @@ void SDL2Mouse::Translate_Event(uint32_t message_num, MouseIO *io)
             break;
         case SDL_MOUSEBUTTONDOWN:
         case SDL_MOUSEBUTTONUP:
-            state = event.type == SDL_MOUSEBUTTONDOWN ? MouseIO::MouseState::MOUSE_STATE_DOWN :
-                                                        MouseIO::MouseState::MOUSE_STATE_UP;
+            doubleClick = event.button.button == m_lastClick.button.button
+                && (event.button.timestamp - m_lastClick.button.timestamp) < Get_Double_Click_Time();
```

Comment:
What happens if you click a button 3 times in quick succession? Will it trigger 2x double click? And if yes, is that desirable?

Reply:
On second look this double click does not look proper. Note that this code runs on both `SDL_MOUSEBUTTONDOWN` and `SDL_MOUSEBUTTONUP`, so either `event.button.button == m_lastClick.button.button` condition is never true, or will will trigger as once you press and release mouse button once.

Reply:
Thanks, good catch!
I changed this to only check for double click on mouse down. However this still replaces the 2nd click by a doubleclick event, which is probably wrong. I want to get more of the surrounding code implemented, so we can properly test this!